src/chart/rendering/drawlegend.cpp

```diff
@@ -28,48 +32,97 @@ void DrawLegend::draw(Gfx::ICanvas &canvas,
     Gen::ChannelId channelType,
     double weight) const
 {
-	auto contentRect =
+	auto markerWindowRect =
 	    style.contentRect(legendLayout, rootStyle.calculatedSize());
 
-	auto &&info = Info{
-	    .canvas = canvas,
-	    .contentRect = contentRect,
+	auto titleRect =
+	    markerWindowRect.popBottom(style.title.getHeight());
+	auto markerWindowHeight = markerWindowRect.height();
+	auto itemHeight = style.label.getHeight();
+	auto markerSize = style.marker.size->get(itemHeight,
+	    style.label.calculatedSize());
+
+	auto fadeHeight = markerSize;
+	markerWindowRect =
+	    markerWindowRect
+	    + Geom::Rect{{0, -fadeHeight}, {0, 2 * fadeHeight}};
+
+	auto fadeElementPercent = fadeHeight / markerWindowRect.height();
+
+	auto &&info = Info{.canvas = canvas,
+	    .titleRect = titleRect,
+	    .markerWindowRect = markerWindowRect,
+	    .fadeHeight = fadeHeight,
+	    .yOverflow = {},
+	    .yOffset = {},
 	    .type = channelType,
 	    .weight = weight,
-	    .itemHeight = style.label.getHeight(),
-	    .titleHeight = style.title.getHeight(),
-	    .markerSize = style.marker.size->get(contentRect.size.y,
-	        style.label.calculatedSize()),
+	    .itemHeight = itemHeight,
+	    .markerSize = markerSize,
 	    .measure = plot->axises.at(channelType).measure,
 	    .dimension = plot->axises.at(channelType).dimension,
-	};
+	    .colorGradientSetter = {markerWindowRect.leftSide(),
+	        Gfx::ColorGradient{{
+	            {0.0, {}},
+	            {fadeElementPercent / 2.0, {}},
+	            {fadeElementPercent, {}},
+	            {1.0 - fadeElementPercent, {}},
+	            {1.0 - fadeElementPercent / 2.0, {}},
+	            {1.0, {}},
```

Comment:
Why do we need 6 points here instead of three?

Reply:
If I understand correctly, this way we create an S curve like gradient function. The goal of the three point (0,0 .5,.27 1,1) was to approximate this curve https://en.wikipedia.org/wiki/Lightness#/media/File:Lightness_approximations.svg

Reply:
On a second tought, we shouldn't even doing that, since background/foreground color could be changed, and for a dark theme we should use an inverse curve. So to be on the safe side, we should have a simple 2 point linear gradient

Reply:
These 6 points represent the whole fade area (upper 3 and lower 3 too) and handle both in the same class.
This is because, between the fade areas, the markers need to be fully colored, and I didn't want to handle differently the 2 sides.

Reply:
So the 

```
egy egyszerű gamma-corrected közelítés:
0% - alpha=1.0
50% - alpha=0,27
100% - alpha=0.0
```
message is irrelevant?

Reply:
OK. I'll remove the 0.27 points.